dll/cpl/sysdm/environment.c

```diff
@@ -17,6 +21,102 @@ typedef struct _VARIABLE_DATA
     LPTSTR lpCookedValue;
 } VARIABLE_DATA, *PVARIABLE_DATA;
 
+typedef struct _DIALOG_DATA
+{
+    BOOL bIsFriendlyUI;
+    DWORD dwSelectedValueIndex;
+    DWORD dwTextEditedValueIndex;
+    HWND hEditBox;
+    PVARIABLE_DATA VarData;
+} DIALOG_DATA, *PDIALOG_DATA;
+
+static DWORD
+GatherDataFromEditBox(HWND hwndDlg,
+                      PVARIABLE_DATA VarData)
+{
+    DWORD dwNameLength;
+    DWORD dwValueLength;
+
+    dwNameLength = (DWORD)SendDlgItemMessage(hwndDlg, IDC_VARIABLE_NAME, WM_GETTEXTLENGTH, 0, 0);
+    dwValueLength = (DWORD)SendDlgItemMessage(hwndDlg, IDC_VARIABLE_VALUE, WM_GETTEXTLENGTH, 0, 0);
```

Comment:
Why certain calls to `SendDlgItemMessage` don't have the type casts but this one?